### DISCUSSION TITLE
bump to engine.io-client 6.6.0?

### DISCUSSION THREAD

**Discussion by Schipy** [socketio/socket.io#5133] [Q&A]
Hi, is there a reason `engine.io-client` is not bumped to `6.6.0`? (in `socket.io-client`). The latest engine.io-client contains bump to `"ws": "~8.17.1"`, which includes the fix for this vulnerability: [CVE-2024-37890](https://nvd.nist.gov/vuln/detail/CVE-2024-37890) ([related WS issue](https://github.com/websockets/ws/issues/2230))

thanks

**Comment by darrachequesne** [comment]
Hi!

`socket.io-client@4.7.5` depends on `engine.io-client@~6.5.2`, which should install `engine.io-client@6.5.4` and thus `ws@8.17.1`. Isn't that the case for you?

**Reply by CsabaSzabo** [reply]
@darrachequesne it works only if the `engine-io-client@6.5.4` is manually resolved, otherwise the `engine.io-client@~6.5.2` installs `engine.io-client@6.5.2` in our project

```
  "resolutions": {
    "engine.io-client": "6.5.4"
  }
```

**Reply by darrachequesne** [reply]
That's weird, manual resolution should not be needed. Which package manager are you using? `npm`? The main point of using `~` versions is to prevent this kind of issues.
